src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -194,6 +190,31 @@ def _validate_target_field(
         default=None,
     )
 
+    mixins: Optional[Dict[str, Dict]] = Field(
+        title="Mixins to apply to entities",
+        default=None,
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def apply_mixins(cls, data: Dict) -> Dict:
+        """
+        Applies mixins to those entities, whose meta field contains the mixin name.
+        """
+        if "mixins" in data and "entities" in data:
+            for entity in data["entities"].values():
+                entity_mixins = entity_mixins_to_list(
+                    entity.get("meta", {}).get("use_mixins")
+                )
+                entity_fields = get_allowed_fields_for_entity(entity)
+                if entity_fields and entity_mixins:
+                    for mixin_name in entity_mixins:
+                        if mixin_name in data["mixins"]:
+                            for key, value in data["mixins"][mixin_name].items():
+                                if key in entity_fields:
+                                    entity[key] = value
```

Comment:
Will this rise an error if we try to set up unknown field?

Reply:
A single mixin can be used to inject data into different entities, this would need some additional logic- possible solution is provided here https://github.com/snowflakedb/snowflake-cli/pull/1484#discussion_r1730784108